Google.Api.Gax.Grpc/ClientBuilder.cs

```diff
@@ -0,0 +1,209 @@
+﻿/*
+ * Copyright 2019 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Apis.Auth.OAuth2;
+using Grpc.Auth;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// Base class for API-specific builders.
+    /// </summary>
+    /// <typeparam name="TClient">The type of client created by this builder.</typeparam>
+    public abstract class ClientBuilder<TClient>
```

Comment:
I *think* I'd rather this was called `ClientBuilderBase`. That's consistent with `ServiceSettingsBase`, `ServerStreamingBase`.

Reply:
Yup, and that's how I remembered it.